src/main/java/org/opensearch/knn/plugin/transport/UpdateModelGraveyardTransportAction.java

```diff
@@ -156,4 +168,72 @@ public ClusterTasksResult<UpdateModelGraveyardTask> execute(ClusterState cluster
             return new ClusterTasksResult.Builder<UpdateModelGraveyardTask>().successes(taskList).build(updatedClusterState);
         }
     }
+
+    private static void checkIfIndicesAreUsingModel(ClusterState clusterState, UpdateModelGraveyardTask task, String[] indicesArray)
+        throws IOException {
+        Map<String, MappingMetadata> mappings = clusterState.metadata()
+            .findMappings(indicesArray, task.getIndicesService().getFieldFilter());
+        Map<String, IndexMetadata> indices = clusterState.metadata().indices();
+        List<String> indicesUsingModel = new ArrayList<>();
+        // Parse indices and add to list if using the model
+        for (Map.Entry<String, MappingMetadata> entry : mappings.entrySet()) {
+            MappingMetadata mappingMetadata = entry.getValue();
+            Settings indexSettings = indices.get(entry.getKey()).getSettings();
+            if (mappingMetadata != null && Boolean.parseBoolean(indexSettings.get("index.knn", "false"))) {
```

Comment:
Would suggest first getting all of the indices with "index.knn" enabled in one loop and then checking the mapping in another.